services/git-service-impl/src/test/java/io/fabric8/launcher/service/gitlab/impl/GitLabServiceIT.java

```diff
@@ -1,28 +1,48 @@
 package io.fabric8.launcher.service.gitlab.impl;
 
-import java.net.URL;
-import java.util.ArrayList;
-import java.util.List;
-import java.util.Optional;
-
 import io.fabric8.launcher.service.git.api.GitHook;
 import io.fabric8.launcher.service.git.api.GitOrganization;
 import io.fabric8.launcher.service.git.api.GitRepository;
 import io.fabric8.launcher.service.git.api.GitUser;
 import io.fabric8.launcher.service.git.spi.GitServiceSpi;
 import io.fabric8.launcher.service.gitlab.api.GitLabService;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
 import org.assertj.core.api.JUnitSoftAssertions;
 import org.junit.After;
+import org.junit.ClassRule;
 import org.junit.Rule;
 import org.junit.Test;
+import org.junit.contrib.java.lang.system.ProvideSystemProperty;
+import org.junit.rules.RuleChain;
+import org.junit.rules.TemporaryFolder;
+import org.junit.rules.TestName;
 
-import static io.fabric8.launcher.service.gitlab.api.GitLabEnvVarSysPropNames.LAUNCHER_MISSIONCONTROL_GITLAB_USERNAME;
+import static io.fabric8.launcher.service.github.test.HoverflyRuleConfigurer.createHoverflyProxy;
 
 /**
  * @author <a href="mailto:ggastald@redhat.com">George Gastaldi</a>
  */
 public class GitLabServiceIT {
 
+    private static final String MY_GITLAB_REPO_DESCRIPTION = "Test project created by Integration Tests.";
+
+    @ClassRule
+    public static RuleChain ruleChain = RuleChain
```

Comment:
Awesome! How can we reuse this set of rules in other tests (like GithubServiceIT)?

Reply:
RuleChain is provided by Junit - https://junit.org/junit4/javadoc/4.12/org/junit/rules/RuleChain.html.

We are using it to make sure we have system properties available to use in another rule. You can chain any Rule which implements `TestRule` interface

Reply:
Right, I think my question was more to: How can we reuse these lines of code to avoid copy pasting in every test that uses Hoverfly. A base class?

Reply:
yes we can do it by writing `@ClassRule` in baseclass and extending it in testclass.